great_expectations/datasource/data_connector/runtime_data_connector.py

```diff
@@ -307,18 +307,14 @@ def build_batch_spec(  # type: ignore[return,override]
         batch_spec: BatchSpec = super().build_batch_spec(
             batch_definition=batch_definition
         )
+        batch_spec.update(runtime_parameters)
+
         if "batch_data" in runtime_parameters:
-            batch_spec["batch_data"] = runtime_parameters.get("batch_data")
             return RuntimeDataBatchSpec(batch_spec)
         elif "query" in runtime_parameters:
-            batch_spec["query"] = runtime_parameters.get("query")
-            batch_spec["temp_table_schema_name"] = runtime_parameters.get(
-                "temp_table_schema_name"
-            )
             return RuntimeQueryBatchSpec(batch_spec)
         elif "path" in runtime_parameters:
             path: str = runtime_parameters["path"]
-            batch_spec["path"] = path
```

Comment:
Just to confirm - are the changes here to allow `create_temp_table` to be passed into the BatchSpec?

If so, would it be possible to instead pass `create_temp_table` via the `batch_spec_passthrough` (e.g. `batch_spec_passthrough={"create_temp_table": True}`) of your ExecutionEngine or RuntimeBatchRequest? This is how we typically add key/values directly to the batch spec.

The concern with the proposal to add all of the `runtime_parameters` to the `batch_spec` is the case where multiple data references are passed would lead to a potentially indeterminate BatchSpec (e.g. batch_data, query and path data references all in the same BatchSpec). We can figure out which reference is correct based on the BatchSpec class, but in the future we may want to be more explicit about which references / properties are acceptable based on the BatchSpec class.

Reply:
Yeah `batch_spec_passthrough`, I like it. 

Currently, I just use the `create_temp_table` parameter, but I would use more parameters in the future. I thought a RuntimeBatchRequest requires all parameters to be passed into `runtime_parameters`. 

Therefore, I am planning to undo the change about `batch_spec` inside `runtime_data_connector`, thank you for the explanation.